pkg/network/tracer/tracer_test.go

```diff
@@ -1072,10 +1070,10 @@ func (s *TracerSuite) TestDNSStats() {
 	cfg.DNSTimeout = 1 * time.Second
```

Comment:
I would not set this for the non-timeout tests to use the default of 15s. A low value here could cause a timeout when we don't want one.

Reply:
You can separate out the timeout test from the other cases and use something even less than `1s` to ensure a timeout.

Reply:
Makes sense, I'd still prefer to leave this for a follow up since:
a) this PR is already large and takes us in the right direction
b) I don't think we've seen instances of successful responses getting classified
c) I want to do a larger refactor of testDNSStats after this PR where splitting the test will fit naturally